### ISSUE TITLE
useEditable does not set scrollPosition for the first Editor render

### ISSUE THREAD

**Issue by dgrechka** [rowsncolumns/grid#7]
When I move the viewport (e.g. with scroll) and then activate the in-cell editor the values passed in `scrollPosition` for `EditorProps` still contains

```javascript
{
  scrollTop: 0;
  scrollLeft: 0;
}
```

instead of actual scroll values. When I then continue scrolling, it immediately gets the correct scroll value.
But the very first render contains wrong zero value of the `scrollPosition`.

Possible cause for this is that `useEditable` hook tracks the scrolling only when the editor is shown (e.g. activeCell is defined).

```typescript
 const handleScroll = useCallback((scrollPos: ScrollCoords) => {
    if (!currentActiveCellRef.current) return;
    setScrollPosition(scrollPos);
  }, []);
```

**Comment by rmdort** [comment]
How do you activate the in-cell editor? When the editor is active, shouldnt `currentActiveCellRef.current` be `true` ?

**Comment by dgrechka** [comment]
I use the `useEditable` hook in the grid component.
Initially the editor is inactive (hidden) as in [the editable grid sample of the storybook](https://rowsncolumns.github.io/grid/?path=/story/grid--editable-grid) (by the way, this sample does not handle scroll).

I don't have active cell yet.

Without selecting any cell, I start scrolling away from grid origin, and when I do this the internal state of `useEditable` is not updated.
```typescript
const [scrollPosition, setScrollPosition] = useState<ScrollCoords>({
    scrollLeft: 0,
    scrollTop: 0,
  });
```
because `currentActiveCellRef.current` is `null`, right?

Then I activate the editor either with double mouse click on some cell or by just starting typing.
As I understand, at that time the scrollPosition still has zeros.
That's why I can't place my Editor properly (calculate its position and account for scroll) to match it with the corresponding cell coordinate.

**Comment by rmdort** [comment]
Hmm.  Are you able to reproduce it here - https://rowsncolumns.github.io/grid/?path=/story/grid--editable-grid

I do grab the scroll position when you activate the editor

https://github.com/rowsncolumns/grid/blob/master/packages/grid/src/hooks/useEditable.tsx#L487

**Comment by dgrechka** [comment]
I see.

So the `position` passed as a prop to Editor already accounts for scroll and it is in canvas coordinates, not in grid coordinates.

That's why the editor spawns at valid location in the https://rowsncolumns.github.io/grid/?path=/story/grid--editable-grid if we first scroll and then activate the editor.

But then I want the editor to be pinned to the cell for which it was spawned when I keep the editor active and start scrolling (the grid moves with a scroll and the editor also moves and keeps itself just over corresponding cell).
I want to account for scroll value here https://github.com/rowsncolumns/grid/blob/master/packages/storybook/src/Grid.stories.tsx#L1032
My intension was to use `scrollPosition`  from `editorProps` here, but first it contains zeros (as it is not tracked in internal state while we have no active cell) and then with a scroll event (when I start scrolling second time, this time I have the editor active and active cell defined) it is updated.
The scroll value which corresponds to the moment of editor activation is missing. I can't calculate the scroll delta to properly calculate the css `left` property for the editor container.

**Comment by rmdort** [comment]
https://github.com/rowsncolumns/grid/blob/master/packages/grid/src/hooks/useEditable.tsx#L493

Would it work if I call `setScrollPosition` in `makeEditable` block? So scrollPosition will be set when u activate editor

**Comment by dgrechka** [comment]
Yes. It will help.
The internal `scrollPosition` will be updated not on the first scroll after the editor is active but on the editor activation. This will make it possible to calculate the scroll delta between latest scroll values and the scroll values of the editor activation moment.
This delta can be used to offset the position.

But personally I would find it more consistent to have both `position` and `scrollPosition ` to be in the same coordinate systems (e.g. not account for scroll in the `position` and force the Rows n' Columns grid user to apply the scroll offset explicitly in css).

For example like this
```jsx
<div
        style={{
          position: "absolute",
          left: position.x - scrollPosition.scrollLeft,
          top: position.y - scrollPosition.scrollTop,
          width: position.width,
          height: position.height
        }}
      >
```

But this preference may differ from user to user. I understand.

I also plan to wrap this div into one more div that does clipping with css `clipPath` property to prevent the opened editor from being rendered over frozen columns and rows. And to cull this floating editor entirely if it moves out of the grid with scrolling.
These cases arise when I start to compensate for the scroll changes after the editor is being active.

**Comment by rmdort** [comment]
Maybe i will add a prop to `useEditable` to support `sticky` editor (the default behavior), If `sticky: false`, then the `position` will not get offset by scrollPosition. Does that sound okay?

~~But ya, you will need to wrap the editor in another `div` for clipping. Let me know whats the best way here~~

To clip the editor on scroll, you could wrap the entire grid in a overflow hidden div.

````
const {editorComponent} = useEditable()
<div style={{position: 'relative', overflow: 'hidden'}}>
  <Grid
  />
  {editorComponent}
<div>
````

**Comment by dgrechka** [comment]
So sticky means that the editor does not move with the scroll. It is pinned to canvas. right?

This handle will be great!

Will the name `reportCanvasCoordinatesPosition` (or `returnCanvasCoordinatesPosition` or `useCanvasCoordinatesPosition`) be more easy to understand for the new users rather then `sticky`. The semantics for these is opposite for the sticky, and thus the default can be `false`.
Although these names can be too long.

Maybe just put JSDoc for the `sticky` with detailed description of the difference.

Thanks for the overflow hidden div hint! That is useful!
it seems I'll still need to clip with `clipPath` if I want to prevent the opened editor to get on top of frozen rows/columns.

Thank you for the great library!
I have being using it for several months and it works perfectly!

**Comment by rmdort** [comment]
Published 8.0.1 to support `sticky` option is `useEditable` hook.

When sticky is true - Editor will always remain on screen during an active edit session.
When sticky is false - Editor will be tied to a cell and will scroll along with the grid.

Here is a demo with clip-path option - https://codesandbox.io/s/staging-flower-yvrhm?file=/src/App.tsx

**Comment by rmdort** [comment]
Closing this issue. Pls reopen if necessary
